src/mailroom.py

```diff
@@ -0,0 +1,339 @@
+               'prompt_message': ' (y or n): ',
+               'valid_responses': state_7_valid_responses,
+               'new_donor_name': ''
+               }
+
+    state_dict = {
+        'state_1': state_1,
+        'state_2': state_2,
+        'state_3': state_3,
+        'state_4': state_4,
+        'state_5': state_5,
+        'state_6': state_6,
+        'state_7': state_7,
+    }
+
+    return state_dict
+
+
+def state_1_valid_responses(a):
+    global CURRENT_STATE
+    global STATE_DICT
+    if a == '0':
+        os.system('clear')
+        print('Good By')
+        sys.exit()
+    elif a == '1':
+        create_report()
+        CURRENT_STATE = STATE_DICT['state_1']
+        return True
+    elif a == '2':
+        CURRENT_STATE = STATE_DICT['state_2']
+        os.system('clear')
+        return True
+    else:
+        print('That was not a valid input')
+        return False
+
+
+def state_2_valid_responses(a):
+    ''' TODO: add check for new donor name greater then 20 characters '''
+    global CURRENT_STATE
+    global CURRENT_DONOR
+    if a == 'l' or a == 'list':
+        CURRENT_STATE = STATE_DICT['state_2']
+        list_of_donors(DONORS)
+    elif a == '0':
+        CURRENT_STATE = STATE_DICT['state_1']
+        os.system('clear')
+    else:
+        try:
+            CURRENT_DONOR = DONORS[a]
+            user_report()
+            CURRENT_STATE = STATE_DICT['state_4']
+        except KeyError:
+            STATE_DICT['state_2']['user_response'] = a
+            print('{} is not a current donor.'.format(a))
+            print('Create a new donor {}'.format(a), end='')
+            STATE_DICT['state_6']['new_donor_name'] = a
+            CURRENT_STATE = STATE_DICT['state_6']
+            return
+
+
+def state_3_valid_responses(a):
+    global CURRENT_STATE
+    if a == '1':
+        CURRENT_STATE = STATE_DICT['state_1']
+        os.system('clear')
+    elif a == '0':
+        os.system('clear')
+        print('Good By')
+        sys.exit()
+    else:
+        print('That was not a valid input')
+        return False
+
+
+def state_4_valid_responses(a):
+    ''' need to test for properly formated $ amout iiiiii.ii '''
+    global CURRENT_STATE
+    global CURRENT_DONOR
+    if a == '0':
+        CURRENT_STATE = STATE_DICT['state_2']
+        os.system('clear')
+        return
+    try:
+        amount = float(a)
+        print('Is ${:,.2f} correct?'.format(amount), end='')
+        STATE_DICT['state_4']['donation_amount'] = amount
+        CURRENT_STATE = STATE_DICT['state_7']
+        return amount
+    except ValueError:
+        print('{} is not a valid amount'.format(a))
+        return False
+
+
+def state_5_valid_responses(a):
+    global CURRENT_STATE
+    if a == 'y':
+        print_letter()
+        CURRENT_STATE = STATE_DICT['state_2']
+    elif a == 'n':
+        os.system('clear')
+        CURRENT_STATE = STATE_DICT['state_2']
+    else:
+        print('That was not a valid input')
```

Comment:
what *is* valid input?  If I got this error message, that would be my first question.